TypeScript JSX ElementType broken in 3.0.x

### Affected Packages

core, react

### Version(s)

3.0.1

### Bug Description

After upgrading from Tiptap 2.26.1 to 3.0.1, my TypeScript React code that uses React's `ElementType` started failing with errors similar to the following:

```
src/Editor.tsx:10:11 - error TS2604: JSX element type 'Component' does not have any construct or call signatures.

10   return <Component />;
             ~~~~~~~~~

src/Editor.tsx:10:11 - error TS2786: 'Component' cannot be used as a JSX component.
  Its type 'React.ElementType' is not a valid JSX element type.
    Type 'number' is not assignable to type 'ElementType'.

10   return <Component />;
             ~~~~~~~~~
```

Here's the code that demonstrates this error:
```
interface Props {
  component: ElementType;
}

export function Test({ component: Component }: Props): ReactElement {
  return <Component />;
}
```

### Browser Used

Chrome

### Code Example URL

https://github.com/joshkel/tiptap-jsx-element-bug

### Expected Behavior

No errors

### Additional Context (Optional)

I believe that the problem is caused by [these types within jsx-runtime.ts](https://github.com/ueberdosis/tiptap/blob/v3.0.0-beta.24/packages/core/src/jsx-runtime.ts#L16):

```ts
declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace JSX {
    // @ts-ignore - conflict with React typings
    type Element = [string, ...any[]]
    // @ts-ignore - conflict with React typings
    interface IntrinsicElements {
      // @ts-ignore - conflict with React typings
      [key: string]: any
    }
  }
}
```

Those are merged with the types provided by @types/react (and throw errors if TypeScript's [`skipLibCheck` option](https://www.typescriptlang.org/tsconfig/#skipLibCheck) isn't enabled), resulting in `ElementType` being something that TypeScript thinks isn't compatible with JSX.

### Dependency Updates

- [x] Yes, I've updated all my dependencies.

--- a/.changeset/grumpy-seals-shake.md
+++ b/.changeset/grumpy-seals-shake.md
@@ -1,0 +1,5 @@
+---
+'@tiptap/core': patch
+---
+
+Fix: Avoid the JSX Runtime to globally overwrite React's Element types when `/** @jsxImportSource @tiptap/core */` is not used

--- a/.changeset/two-chefs-lie.md
+++ b/.changeset/two-chefs-lie.md
@@ -1,0 +1,6 @@
+---
+'@tiptap/extension-blockquote': patch
+'@tiptap/extension-bold': patch
+---
+
+Chore: Removed type overwrites for renderHTML

--- a/packages/core/src/jsx-runtime.ts
+++ b/packages/core/src/jsx-runtime.ts
@@ -13,16 +13,17 @@ export type DOMOutputSpecArray =
   | [string, Attributes, DOMOutputSpecArray | 0]
   | [string, DOMOutputSpecArray]
 
-declare global {
-  // eslint-disable-next-line @typescript-eslint/no-namespace
-  namespace JSX {
-    // @ts-ignore - conflict with React typings
-    type Element = [string, ...any[]]
-    // @ts-ignore - conflict with React typings
-    interface IntrinsicElements {
-      // @ts-ignore - conflict with React typings
-      [key: string]: any
-    }
+// JSX types for Tiptap's JSX runtime
+// These types only apply when using @jsxImportSource @tiptap/core
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace JSX {
+  export type Element = DOMOutputSpecArray
+  export interface IntrinsicElements {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    [key: string]: any
+  }
+  export interface ElementChildrenAttribute {
+    children: unknown
   }
 }
 

--- a/packages/extension-blockquote/src/blockquote.tsx
+++ b/packages/extension-blockquote/src/blockquote.tsx
@@ -1,6 +1,5 @@
 /** @jsxImportSource @tiptap/core */
 import { mergeAttributes, Node, wrappingInputRule } from '@tiptap/core'
-import type { DOMOutputSpecArray } from '@tiptap/core/jsx-runtime'
 
 export interface BlockquoteOptions {
   /**
@@ -63,7 +62,7 @@ export const Blockquote = Node.create<BlockquoteOptions>({
       <blockquote {...mergeAttributes(this.options.HTMLAttributes, HTMLAttributes)}>
         <slot />
       </blockquote>
-    ) as unknown as DOMOutputSpecArray
+    )
   },
 
   addCommands() {

--- a/packages/extension-bold/src/bold.tsx
+++ b/packages/extension-bold/src/bold.tsx
@@ -1,6 +1,5 @@
 /** @jsxImportSource @tiptap/core */
 import { Mark, markInputRule, markPasteRule, mergeAttributes } from '@tiptap/core'
-import type { DOMOutputSpecArray } from '@tiptap/core/jsx-runtime'
 
 export interface BoldOptions {
   /**
@@ -88,7 +87,7 @@ export const Bold = Mark.create<BoldOptions>({
       <strong {...mergeAttributes(this.options.HTMLAttributes, HTMLAttributes)}>
         <slot />
       </strong>
-    ) as unknown as DOMOutputSpecArray
+    )
   },
 
   addCommands() {

